src/PagerView.tsx

```diff
@@ -1,155 +1,207 @@
+    super(props);
+    this.state = this.computeRenderWindow({
+      buffer: props.buffer,
+      currentPage: props.initialPage ?? 0,
+      maxRenderWindow: props.maxRenderWindow,
+      offset: 0,
+      windowLength: 0,
+    });
+  }
 
-  private _onPageSelected = (e: PagerViewOnPageSelectedEvent) => {
-    if (this.props.onPageSelected) {
-      this.props.onPageSelected(e);
+  componentDidMount() {
+    if (this.props.initialPage != null && this.props.initialPage > 0) {
+      // Send command directly; render window already contains destination.
+      UIManager.dispatchViewManagerCommand(findNodeHandle(this), 'setPage', [
+        this.props.initialPage,
+        false,
+      ]);
     }
-  };
+  }
 
   /**
    * A helper function to scroll to a specific page in the PagerView.
-   * The transition between pages will be animated.
+   * Default to animated transition between pages.
    */
-  public setPage = (selectedPage: number) => {
-    UIManager.dispatchViewManagerCommand(
-      ReactNative.findNodeHandle(this),
-      getViewManagerConfig().Commands.setPage,
-      [selectedPage]
+  setPage(page: number, animated = true) {
+    if (page < 0 || page >= this.props.data.length) {
+      return;
+    }
+
+    // Start rendering the destination.
+    this.setState((prevState) =>
+      this.computeRenderWindow({
+        buffer: this.props.buffer,
+        currentPage: page,
+        maxRenderWindow: this.props.maxRenderWindow,
+        offset: prevState.offset,
+        windowLength: prevState.windowLength,
+      })
     );
-  };
+    // Send paging command.
+    requestAnimationFrame(() => {
+      UIManager.dispatchViewManagerCommand(findNodeHandle(this), 'setPage', [
+        page,
+        animated,
+      ]);
+    });
+  }
 
   /**
    * A helper function to scroll to a specific page in the PagerView.
    * The transition between pages will *not* be animated.
    */
-  public setPageWithoutAnimation = (selectedPage: number) => {
-    UIManager.dispatchViewManagerCommand(
-      ReactNative.findNodeHandle(this),
-      getViewManagerConfig().Commands.setPageWithoutAnimation,
-      [selectedPage]
-    );
-  };
+  setPageWithoutAnimation(page: number) {
+    this.setPage(page, false);
+  }
 
   /**
-   * A helper function to enable/disable scroll imperatively
-   * The recommended way is using the scrollEnabled prop, however, there might be a case where a
-   * imperative solution is more useful (e.g. for not blocking an animation)
+   * A helper function to enable/disable scroll imperatively.
+   * The recommended way is using the scrollEnabled prop, however, there might
+   * be a case where an imperative solution is more useful (e.g. for not
+   * blocking an animation)
    */
-  public setScrollEnabled = (scrollEnabled: boolean) => {
+  setScrollEnabled(scrollEnabled: boolean) {
     UIManager.dispatchViewManagerCommand(
-      ReactNative.findNodeHandle(this),
-      getViewManagerConfig().Commands.setScrollEnabled,
```

Comment:
Why `getViewManagerConfig().Commands.setScrollEnabled,` has been changed to `setScrollEnabled`